### PR TITLE
Add support for building against Java 11

### PR DESCRIPTION
**Pull request by eric-weaver**
- Adds a new maven profile that is automatically activated if current JAVA_HOME jdk is java 11
- Adds openjdk11 build to travis
- Moves all dependency versions to property variables
- Updates the maven-shade-plugin dependency
- Adds the maven-enforcer-plugin to set minimum maven version
 - Removed cobertura plugin. Wasn't configured to run previously. It also receives a class not found issue on java 11

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -21,138 +21,204 @@
...
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.mockito</groupId>
             <artifactId>mockito-core</artifactId>
-            <version>2.25.1</version>
+            <version>${version.mockito}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (arron-green) [inline pom.xml:177]: did you mean to bring this one in? might want to move it bear the runtime dependencies.  Currently it's listed with the `test` dependencies

Comment (eric-weaver) [reply pom.xml:177]: This is required for java 11

File: pom.xml [hunk 2]
```diff
@@ -21,138 +21,204 @@
...
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>${version.compiler.plugin}</version>
+                <configuration>
+                    <source>${maven.compiler.source}</source>
+                    <target>${maven.compiler.target}</target>
+                    <encoding>${project.build.sourceEncoding}</encoding>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
```
Comment (arron-green) [inline pom.xml:199]: 👍  this could help with enforcing some of the transitive dependency issues this project currently has

**Review by arron-green — APPROVED** [review]
nice work!

**Pull request merged**